### PR TITLE
[Bug] block input while waiting for transition to prevent changing starter during move select

### PR DESCRIPTION
**Pull request by scottarrr**
## What are the changes?

This uses the existing blockInput flag to prevent inputs from changing the starter when opening the manage moves menu.

## Why am I doing these changes?

addresses these copies of the same bug
https://github.com/pagefaultgames/pokerogue/issues/1879
https://github.com/pagefaultgames/pokerogue/issues/2283

## What did change?

Previously inputs that happened in between the triggering of swap to options mode and before it completed would be handled by the starter select, changing the starter selection. 

### Screenshots/Videos

a video wouldn't really prove that i'm hitting the buttons as described in the bug ;/

## How to test the changes?

Pull this down, open starter select, pick bulbasaur and very quickly input a right movement action.
Previously bulbasaur's moveset would open with charmander selected
With this branch bulbasaur's moves are shown and bulbasaur remains the selection

## Checklist
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [ ] Are the changes visual?
  - [ ] Have I provided screenshots/videos of the changes?

### CODE REVIEW

File: src/ui/starter-select-ui-handler.ts [hunk 1]
```diff
@@ -1149,6 +1149,9 @@ export default class StarterSelectUiHandler extends MessageUiHandler {
           ];
           if (this.speciesStarterMoves.length > 1) {
             const showSwapOptions = (moveset: StarterMoveset) => {
+
+              this.blockInput = true;
```
Comment (brain-frog) [inline src/ui/starter-select-ui-handler.ts:1153]: can you add this to the manage nature menu handler as well so the bug doesnt happen there too

**Review by brain-frog — DISMISSED** [review]

File: src/ui/starter-select-ui-handler.ts (hunk 1 above)
Comment (scottarrr) [reply src/ui/starter-select-ui-handler.ts:1153]: Absolutely. Good call!

**Comment by scottarrr** [comment]
@brain-frog @f-fsantos Sorry for the ping, I see the base branch was swapped to beta but I'm not all that familiar with the process here. Is there anything waiting on me?

**Comment by scottarrr** [comment]
@Tempo-anon  Is this still needed? I see the bugs were closed so lmk if I should close this out too.

**Pull request merged**